src/librustc/middle/const_eval.rs

```diff
@@ -1265,12 +1273,8 @@ fn resolve_trait_associated_const<'a, 'tcx: 'a>(tcx: &'a ty::ctxt<'tcx>,
         Ok(None) => {
             return None
         }
-        Err(e) => {
-            tcx.sess.span_bug(ti.span,
-                              &format!("Encountered error `{:?}` when trying \
-                                        to select an implementation for \
-                                        constant trait item reference.",
-                                       e))
+        Err(_) => {
+            return None
```

Comment:
IIRC this was to avoid hitting an error in early const eval. `fn select()` returns `Err(_)` if there's a substitution that can't be found. Maybe it should return `Ok(None)`?